### PR TITLE
Fix/loopy changes

### PR DESCRIPTION
**Pull request by wence-**
Ports the custom kernels to the new kernel interface. Although we can still use string kernels, one level of indirection has been removed. I also correct some of the access descriptors.

Needs firedrakeproject/firedrake#1408 to land simultaneously.

### CODE REVIEW

**Comment by wence-** [comment]
The test suite passes for me locally with these changes and the relevant firedrake etc... branches.

File: gusto/recovery.py [hunk 1]
```diff
@@ -163,91 +167,91 @@ def __init__(self, v0, v1, v_out, method='density'):
...
-            float y = COORDS[1][1] - COORDS[0][1];
-            float a = CG1[3][0];
-            float b = CG1[1][0];
-            float c = DG0[0][0];
-            DG1[1][0] = a;
-            DG1[3][0] = b;
-            DG1[2][0] = (1.0 / (pow(x, 2.0) + 4.0 * pow(y, 2.0))) * (-3.0 * a * pow(y, 2.0) - b * pow(x, 2.0) - b * pow(y, 2.0) + 2.0 * c * pow(x, 2.0) + 8.0 * c * pow(y, 2.0));
-            DG1[0][0] = 4.0 * c - b - a - DG1[2][0];
+            float x = COORDS[2*2] - COORDS[0];
+            float y = COORDS[2*1 + 1] - COORDS[1];
```
Comment (tommbendall) [inline gusto/recovery.py]: To make sure I've got this right, for vector function `f` in spatial dimension `d` we have `f[a][b] -> f[a*d + b]` ?

Comment (wence-) [reply gusto/recovery.py]: Yes.

**Comment by wence-** [comment]
Testing on Jenkins, I hope it will be happy and will then hit go on Monday.

**Pull request merged**